Command line deployment to Azure and deployment scenario visibility

CamSoper opened the pull request:
Addresses https://github.com/dotnet/docs/issues/2716

[Review URL](https://review.docs.microsoft.com/en-us/aspnet/core/tutorials/publish-to-azure-webapp-using-cli?branch=pr-en-us-4728)

dnfclas commented:
@__CamSoper__,
Thanks for your contribution.
To ensure that the project team has proper rights to use your work, please complete the Contribution License Agreement at https://cla2.dotnetfoundation.org.

It will cover your contributions to all .NET Foundation-managed open source projects.
_Thanks,
.NET Foundation Pull Request Bot_

CamSoper commented:
Hey @Rick-Anderson @rachelappel @scottaddie , which GitHub team do I need to be on to not get the cla-required warning, and change labels, etc.?

Also, does this repo not automatically stage the docs anywhere for review?  I'm not done with the PR yet, but I wanted to get this article on a staging server before I started making changes to the TOC or add the deployment links to other tutorials that Mads requested.

Rick-Anderson commented:
@Eilon  can you add @CamSoper  to asp.net so he doesn't get the CLA -error?

Rick-Anderson commented:
@CamSoper  I think we need to get this is the TOC too.

Rick-Anderson commented on aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md:
````diff
@@ -0,0 +1,149 @@
+```cmd
+REM Create the directory and navigate to it.
+REM These commands are Windows-specific. The Bash equivalents are mkdir and cd.
+md MyApplication
````
prefer `dotnet new razor -o aspnetcoreapp`
use Razor, not MVC and create the output dir with the command

Rick-Anderson commented on aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md:
```diff
@@ -0,0 +1,149 @@
+
+![Command line output](publish-to-azure-webapp-using-cli/_static/new_prj.png)
+
+You can then test the application by browsing to the displayed address (usually `http://localhost:5000`).
```
I've pretty much banned "you" and it's forms from this repo. How about
Test the app by browsing to http://localhost:5000
Notes:
I think it's always 5000
Don't code fence it, that way they can click on the link

Rick-Anderson commented on aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md:
````diff
@@ -0,0 +1,149 @@
+az webapp create --name $webappname --resource-group DotNetAzureTutorial --plan $webappname
+```
+
+Before you deploy, you need to set the account-level deployment credentials.  Use the script below, making sure to include your own values for the user name and password.
````
Before deployment, set the account-level deployment credentials using the following script:

Or,

Before deployment, set the account-level deployment credentials using the following script. Replace the "<desired user name>" and ... 

But I don't think we need to tell them.

CamSoper commented:
@Rick-Anderson I agree.  I just hadn't gotten that far yet.  I've got the change to the TOC and some visibility links in other tutorials (e.g., Next Steps:  Deploy your application using VS or the CLI) I'm still working on.  :)

Rick-Anderson commented on aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md:
````diff
@@ -0,0 +1,149 @@
+```azurecli-interactive
+az webapp deployment source config-local-git -n $webappname -g DotNetAzureTutorial --query [url] -o tsv
+```
+Note the displayed URL ending in `.git`.  You'll need it in the next step.
````
Total ban on future tense.
You need it in the next step. But better yet, get rid of you.

Note the displayed URL ending in `.git`; it's used it in the next step.

Rick-Anderson commented on aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md:
```diff
@@ -0,0 +1,149 @@
+
+## Deploy the application using Git
+
+You're ready to deploy from your local machine using Git.
```
The app is ready to deploy ...

Rick-Anderson commented on aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md:
````diff
@@ -0,0 +1,149 @@
+git push azure master
+```
+
+You will be prompted for the deployment credentials you set earlier.  After authenticating, the application will be built remotely and deployed.
````
no future tense

Eilon commented:
@Rick-Anderson / @CamSoper - there's no magic list, you have to sign the CLA.

dnfclas commented:
__@CamSoper__, thanks for signing the contribution license agreement. We will now validate the agreement and then the pull request.
<br />_Thanks, .NET Foundation Pull Request Bot_

Eilon commented:
There ya go 😄

CamSoper commented:
@Rick-Anderson Requested changes made, added to TOC, added some text and a few links to increase visibility.  We've got publishing to Azure mentioned now at the end of every tutorial except the EF tutorials (didn't seem like a great fit), but the EF tutorials do link to the publishing page.  The "Advanced Topics" EF page might benefit from a bit about publishing to Azure, but it's going to be a tiny bit complicated by the addition of SQL Database.

CamSoper commented:
@Rick-Anderson Is there something special about how the TOC works in your repo?  The new TOC entry is showing up, but it's not clickable, nor is it highlighted when viewing the new article...

Rick-Anderson commented:
@CamSoper  looks great!
